common.mk

```diff
@@ -51,7 +51,10 @@ WITH_CXX_TESTS			?= y
 endif
 endif
 
-CCACHE ?= $(shell which ccache) # Don't remove this comment (space is needed)
+CCACHE_present := $(shell which ccache)
+ifneq ($(CCACHE_present),)
+CCACHE ?= $(CCACHE_present) # Don't remove this comment (space is needed)
```

Comment:
What problem are you trying to solve with this?

Reply:
Hi Jens, TF-A v2.11 introduced some build system hardening implying CROSS_COMPILE must contain the exact path to the toolchain. For the case when ccache is not installed, this statement:
https://github.com/OP-TEE/build/blob/b80d8cb05ef6b789e536631f6e8d88118e8483b4/common.mk#L195
breaks TF-A build because CROSS_COMPILE gets an extra heading space character. TF-A attempts invoking $(CROSS_COMPILE)gcc but but resolves to 
CROSS_COMPILE=" <...>/build/../toolchains/aarch64/bin/aarch64-linux-gnu-"  instead of
CROSS_COMPILE="<...>/build/../toolchains/aarch64/bin/aarch64-linux-gnu-"

Reply:
Thanks for the detailed explanation, please add a short comment in the makefile to help others understand what's happening. Perhaps:
```
# Only set CCACHE if it's pointing to something to avoid prefixing CROSS_COMPILE
# with whitespace. TF-A will not build with whitespace first in CROSS_COMPILE.
```

Reply:
Done.